torch/distributed/tensor/parallel/style.py

```diff
@@ -370,31 +341,28 @@ def make_input_redistribute_1d(
     return functools.partial(make_input_redistribute_1d, input_layouts, output_layouts)
 
 
+def _redistribute_per_layout(layout, use_local_output, t, device_mesh):
+    dst_placements = (layout,)
+    if isinstance(t, DTensor):
+        dtensor = t.redistribute(device_mesh, dst_placements)
+        return dtensor.to_local() if use_local_output else dtensor
+    else:
+        if layout is not None:
+            raise RuntimeError(
+                "Tensor parallel module expects DTensor or tensor"
+                f" when layout specified but received {type(t)}!"
+            )
+        else:
+            return t
+
+
 def _get_prepare_output(
```

Comment:
can we delete those code? Looks like they will not be used.

Reply:
They are used. Synced offline, will move them inside `PrepareModuleOutput` and `PrepareModuleInput`